crates/benda/src/types/u24.rs

```diff
@@ -0,0 +1,91 @@
+use std::ops::{Add, Sub};
+
+use bend::imp;
+use pyo3::{pyclass, pymethods};
+
+use super::BendType;
+
+#[pyclass(module = "benda")]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct u24(u32);
+
+impl BendType for u24 {
+    fn to_bend(&self) -> bend::imp::Expr {
+        imp::Expr::Num {
+            val: bend::fun::Num::U24(self.0),
+        }
+    }
+}
+
+impl u24 {
+    const MAX: u32 = 0xffffff;
+
+    pub fn new(value: u32) -> Self {
+        Self(value & Self::MAX)
+    }
+
+    pub fn get(self) -> u32 {
+        self.0
+    }
+}
+
+impl std::fmt::Debug for u24 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl std::fmt::Display for u24 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<u32> for u24 {
+    fn from(value: u32) -> Self {
+        u24::new(value)
+    }
+}
+
+impl From<u24> for u32 {
+    fn from(val: u24) -> Self {
+        val.0
+    }
+}
+
+impl std::ops::Add for u24 {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        u24::new(self.0 + other.0)
+    }
+}
+
+impl std::ops::Sub for u24 {
+    type Output = Self;
+
+    fn sub(self, other: Self) -> Self {
+        u24::new(self.0 - other.0)
+    }
+}
+
+#[pymethods]
+impl u24 {
+    #[new]
+    fn new_py(value: u32) -> Self {
+        u24::new(value)
+    }
```

Comment:
Add TODO comments for properly implementing the operations (masking, overflow etc) and tests.

We should have TODO comments in the points on the code we already know are lacking something.

Reply:
The u24, f24 and i24 are proof of concept for now. I'll add the TODOs.